src/sentry/dynamic_sampling/tasks/recalibrate_orgs.py

```diff
@@ -61,109 +40,102 @@ def __init__(self, org_id, message):
-            # the query triggering this job and the actual execution of the job.
-            organization = None
-
-        # By default, we use the blended sample rate.
-        target_sample_rate = quotas.backend.get_blended_sample_rate(
-            organization_id=org_volume.org_id
-        )
+            if org_volume.is_valid_for_recalibration:
+                valid_orgs.append((org_volume.org_id, org_volume.total, org_volume.indexed))
+            else:
+                log_recalibrate_org_error(
+                    org_volume.org_id, "The organization is not valid for recalibration"
+                )
+
+        # We run an asynchronous job for recalibrating a batch of orgs whose size is specified in
+        # `GetActiveOrgsVolumes`.
+        recalibrate_orgs_batch.delay(valid_orgs)
 
-        # If we have the sliding window org enabled, we use that and fall back to the blended sample rate in case
-        # of issues.
-        if organization is not None and is_sliding_window_org_enabled(organization):
-            target_sample_rate = get_sliding_window_org_sample_rate(
-                org_id=org_volume.org_id,
-                default_sample_rate=target_sample_rate,
-                notify_missing=True,
-            )
-            log_sample_rate_source(
-                org_volume.org_id,
-                None,
-                "recalibrate_orgs",
-                "sliding_window_org",
-                target_sample_rate,
-            )
-        else:
-            log_sample_rate_source(
-                org_volume.org_id,
-                None,
-                "recalibrate_orgs",
-                "blended_sample_rate",
-                target_sample_rate,
-            )
-
-        # If we didn't find any sample rate, we can't recalibrate the organization.
-        if target_sample_rate is None:
-            raise RecalibrationError(
-                org_id=org_volume.org_id,
-                message="Couldn't get target sample rate for org recalibration",
-            )
-
-        # We compute the effective sample rate that we had in the last considered time window.
-        effective_sample_rate = org_volume.indexed / org_volume.total
-        # We get the previous factor that was used for the recalibration.
-        previous_factor = get_adjusted_factor(org_volume.org_id)
-
-        log_recalibrate_org_state(
-            org_volume.org_id, previous_factor, effective_sample_rate, target_sample_rate
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.recalibrate_orgs_batch",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=25 * 60,
+    time_limit=2 * 60 + 5,
+    silo_mode=SiloMode.REGION,
+)
+def recalibrate_orgs_batch(orgs: Sequence[tuple[int, int, int]]) -> None:
+    for org_id, total, indexed in orgs:
+        recalibrate_org(org_id, total, indexed)
+
+
+def recalibrate_org(org_id: int, total: int, indexed: int) -> None:
```

Comment:
Same code as before, except we do not raise exceptions but rather we log and early return.